### DISCUSSION TITLE
Ubuntu 20.04 LTS - SO 2.3.110 - Most Docker containers "WAIT_START" or "STARTING"

### DISCUSSION THREAD

**Discussion by jonvassmer** [Security-Onion-Solutions/securityonion#7615] [Unsupported Versions] [answered]
Need help in how to troubleshoot the containers not starting up. Only two completely startup (  so-strelka-coordinator and  so-strelka-gatekeeper) all others are in the above mentioned state. This is after about 12 hours, after letting it run over night.
Any help or ideas would be appreciated.

![image](https://user-images.githubusercontent.com/24372498/159146114-a01cfe0a-15f3-415d-8ee4-d8876371fa47.png)
![Capture](https://user-images.githubusercontent.com/24372498/159167597-b2540f25-7d14-4c4d-a527-d8be9cd2df0e.PNG)

**Comment by dougburks** [comment, marked as answer]
Ubuntu 20.04 works fine for me:
```
doug@ubuntu:~$ sudo so-status

Checking Docker status

    Docker -------------------------------------- [ OK ]    

Checking container statuses

    so-aptcacherng ------------------------------ [ OK ]    
    so-curator ---------------------------------- [ OK ]    
    so-dockerregistry --------------------------- [ OK ]    
    so-elastalert ------------------------------- [ OK ]    
    so-elasticsearch ---------------------------- [ OK ]    
    so-filebeat --------------------------------- [ OK ]    
    so-fleet ------------------------------------ [ OK ]    
    so-grafana ---------------------------------- [ OK ]    
    so-idstools --------------------------------- [ OK ]    
    so-influxdb --------------------------------- [ OK ]    
    so-kibana ----------------------------------- [ OK ]    
    so-kratos ----------------------------------- [ OK ]    
    so-logstash --------------------------------- [ OK ]    
    so-mysql ------------------------------------ [ OK ]    
    so-nginx ------------------------------------ [ OK ]    
    so-playbook --------------------------------- [ OK ]    
    so-redis ------------------------------------ [ OK ]    
    so-sensoroni -------------------------------- [ OK ]    
    so-soc -------------------------------------- [ OK ]    
    so-soctopus --------------------------------- [ OK ]    
    so-steno ------------------------------------ [ OK ]    
    so-strelka-backend -------------------------- [ OK ]    
    so-strelka-coordinator ---------------------- [ OK ]    
    so-strelka-filestream ----------------------- [ OK ]    
    so-strelka-frontend ------------------------- [ OK ]    
    so-strelka-gatekeeper ----------------------- [ OK ]    
    so-strelka-manager -------------------------- [ OK ]    
    so-suricata --------------------------------- [ OK ]    
    so-telegraf --------------------------------- [ OK ]    
    so-wazuh ------------------------------------ [ OK ]    
    so-zeek ------------------------------------- [ OK ]    
```

What options did you choose during Setup?

What are the hardware specs of your machine?

**Reply by jonvassmer** [reply]
Doug, 
First, I appreciate your time, response, and the work you do/have done for this project. 
This is a fresh install of Ubuntu and SO. I feel I followed the doc pretty well. and chose only default options.
 They only caveat, is I created the /nsm partition on a separate internal ssd drive. Would this cause any issues? 

Spec's:
 *-core
       description: Motherboard
       product: Z370M-ITX/ac
       vendor: ASRock
       physical id: 0

 *-cpu
          description: CPU
          product: Intel(R) Core(TM) i5-8400 CPU @ 2.80GHz
          vendor: Intel Corp.
          version: Intel(R) Core(TM) i5-8400 CPU @ 2.80GHz
          size: 3874MHz
          capacity: 4005MHz
          width: 64 bits

 *-memory
          description: System Memory
          physical id: a
          slot: System board or motherboard
          size: 16GiB
-1TB ssd for /nsm
-250gb ssd for Ubuntu and SO

Also, are there any logs that I can investigate as  to  the reason why the rest of the images seem to hang? 

Again, thanks for any suggestions or help

**Reply by DebianGuru** [reply]
I actually just had this happen yesterday.  The machine had been working for some time.   I tried for example to run "so-curator-start" because it was in a WAIT_FAILED state and I'd see "Failed to pull securityonion:5000//so-curator:HH1.2.2: Error 400: invalid reference format" being returned from salt.  I saw thing like "[CRITICAL] Rendering SLS 'base:elastalert' failed: while parsing a block mapping" in the syslog.  

Turns out that I accidentally added a character to the beginning of my global.sls file when I must have been editing it.  It was causing the whole file to not be read correctly and halting the whole system.

**Reply by jonvassmer** [reply]
Thanks for the hint. It looks like it maybe a permissions issue:
Mar 24 20:18:44 elk salt-minion[1053718]: [ERROR   ] Container 'so-suricata' is already configured as specified. Failed to start container 'so-suricata': 'Error 500: error while creating mount source path '/opt/so/conf/suricata/bpf': mkdir /opt/so: read-only file system'.
Mar 24 20:18:46 elk salt-minion[1053718]: [ERROR   ] Container 'so-zeek' is already configured as specified. Failed to start container 'so-zeek': 'Error 500: error while creating mount source path '/opt/so/conf/zeek/node.cfg': mkdir /opt/so: read-only file system'.
Mar 24 20:19:02 elk salt-minion[1053718]: [ERROR   ] Container 'so-strelka-frontend' is already configured as specified. Failed to start container 'so-strelka-frontend': 'Error 500: error while creating mount source path '/nsm/strelka/log': mkdir /nsm: read-only file system'.
Mar 24 20:19:04 elk salt-minion[1053718]: [ERROR   ] Container 'so-strelka-backend' is already configured as specified. Failed to start container 'so-strelka-backend': 'Error 500: error while creating mount source path '/opt/so/conf/strelka/backend': mkdir /opt/so: read-only file system'.
Mar 24 20:19:05 elk salt-minion[1053718]: [ERROR   ] Container 'so-strelka-manager' is already configured as specified. Failed to start container 'so-strelka-manager': 'Error 500: error while creating mount source path '/opt/so/conf/strelka/manager': mkdir /opt/so: read-only file system'.
Mar 24 20:19:07 elk salt-minion[1053718]: [ERROR   ] Container 'so-strelka-filestream' is already configured as specified. Failed to start container 'so-strelka-filestream': 'Error 500: error while creating mount source path '/nsm/strelka': mkdir /nsm: read-only file system'.
Mar 24 20:19:09 elk salt-minion[1053718]: [ERROR   ] Container 'so-filebeat' is already configured as specified. Failed to start container 'so-filebeat': 'Error 500: error while creating mount source path '/opt/so/conf/filebeat/modules': mkdir /opt/so: read-only file system'.
Mar 24 20:19:12 elk salt-minion[1053718]: [ERROR   ] {'container_id': {'added': 'e11374a9968e9d984cc9438e8d10bf9fa248ce71c495dcebf989ca9fde6f0763'}}
Mar 24 20:20:22 elk salt-master[1122]: [ERROR   ] Event iteration failed with exception: 'list' object has no attribute 'items'
root@elk:/home/jon# **ls -lath /opt**
total 20K
drwxr-xr-x 10 socore socore 4.0K Mar 24 20:09 so
drwxr-xr-x  5 root   root   4.0K Mar 19 13:40 .
drwxr-xr-x  7 socore socore 4.0K Mar 19 13:03 so_old_1647697247
drwx--x--x  4 root   root   4.0K Mar 19 13:02 containerd

**Reply by jonvassmer** [reply]
removed and then reinstalled docker according to dockers site and all seems to be working now. Thank you for your time and help.
